executor/cook/progress.py

```diff
@@ -231,41 +252,50 @@ def track_progress(progress_watcher, progress_complete_event, send_progress_upda
-                                          progress_updater.send_progress_update))
-    progress_update_thread.start()
-    return progress_complete_event
+    def __init__(self, task_id, config, stop_signal, task_completed_signal, send_progress_message, location, counter):
+        """Launches the threads that track progress and send progress updates to the driver.
 
+        Parameters
+        ----------
+        driver: MesosExecutorDriver
+            The mesos driver to use.
+        task_id: string
+            The task id.
+        config: cook.config.ExecutorConfig
+            The current executor config.
+        stop_signal: Event
+            Event that determines if an interrupt was sent
+        task_completed_signal: Event
+            Event that tracks task execution completion
+        send_progress_message: function(message)
+            The helper function used to send the progress message
+        location: string
+            The target location to read for progress messages
+        counter: ProgressSequenceCounter
+            The sequence counter."""
+        self.progress_complete_event = Event()
+        self.watcher = ProgressWatcher(location, counter, config.max_bytes_read_per_line, config.progress_regex_string,
+                                       stop_signal, task_completed_signal)
+        self.updater = ProgressUpdater(task_id, config.max_message_length, config.progress_sample_interval_ms,
+                                       send_progress_message)
 
-def force_send_progress_update(progress_watcher, progress_updater):
-    """Retrieves the latest progress message and attempts to force send it to the scheduler.
+    def start(self):
+        """Launches a thread that starts monitoring the progress location for progress messages."""
+        logging.info('Starting progress monitoring from {}'.format(self.watcher.progress_target_file()))
+        progress_complete_event = self.progress_complete_event
+        send_progress_update_fn = self.updater.send_progress_update
+        Thread(target=track_progress, args=(self.watcher, progress_complete_event, send_progress_update_fn)).start()
 
-    Parameters
-    ----------
-    progress_watcher: ProgressWatcher
-        The progress watcher which maintains the current progress state.
-    progress_updater: ProgressUpdater
-        The progress updater which sends progress updates to the scheduler.
-        
-    Returns
-    -------
-    Nothing.
-    """
-    latest_progress = progress_watcher.current_progress()
-    progress_updater.send_progress_update(latest_progress, force_send=True)
+    def wait(self, timeout=None):
+        """Waits for the progress tracker thread to run to completion."""
+        self.progress_complete_event.wait(timeout=timeout)
+        if self.progress_complete_event.isSet():
+            logging.info('Progress monitoring from {} complete'.format(self.watcher.progress_target_file()))
```

Comment:
A log message would also be useful if the event *wasn't* set.

Reply:
Okay, will add

Reply:
added log message